res/controllers/Hercules-DJControl-Inpulse-300-script.js

```diff
@@ -44,76 +60,239 @@ DJCi300.scratchShiftMultiplier = 4;
 DJCi300.bendScale = 1.0;
 
 // Other scratch related options
-DJCi300.kScratchActionNone = 0;
+DJCi300.kScratchActionBend = 0;
 DJCi300.kScratchActionScratch = 1;
 DJCi300.kScratchActionSeek = 2;
-DJCi300.kScratchActionBend = 3;
+
+// Pad modes
+DJCi300.padModeNone = -1;
+DJCi300.padModeHotcue = 0;
+DJCi300.padModeRoll = 1;
+DJCi300.padModeSlicer = 2;
+DJCi300.padModeSampler = 3;
+DJCi300.padModeToneplay = 4;
+DJCi300.padModeFX = 5;
+DJCi300.padModeSlicerloop = 6;
+DJCi300.padModeBeatjump = 7;
+
+// Slicer connections
+let slicerSizeConnection1;
```

Comment:
all variables must be part of the DJCi300 object to prevent possible collisions in javascript

Reply:
The only suggestion I would like to make here is to put these into an array rather than postfixing the variables with numbers:
```suggestion
const slizerConnections = {
   size: [undefined, undefined],
   beat: [undefined, undefined],
   track: [undefined, undefined],
}
```